### PR TITLE
ipopt: 3.12.13 -> 3.14.5

### LINKED ISSUE #161858: ipopt package is missing its binary
### Describe the bug

The `ipopt` package doesn't include a `bin/` folder. I am looking for a version of `ipopt` which includes the executable. Are we building the package incorrectly such that it doesn't include a binary?

Any and all help/advice is much appreciated :bow:

### Steps To Reproduce
1. `nix-build -A ipopt`
2. inspect output - notice there is no `bin/` folder

### Expected behavior

I want the `ipopt` executable.

### Notify maintainers

@abbradar
@markuskowa might know something about this? :man_shrugging:

### PR DESCRIPTION
**Pull request by aanderse**
###### Motivation for this change
- update package
- build `ipopt` executable - resolves #161858

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by markuskowa — COMMENTED** [review]
There's a bit more of a rat's tail involved, since the update also affects `pagmo2` and `nlopt`:
```
diff --git a/pkgs/development/libraries/nlopt/default.nix b/pkgs/development/libraries/nlopt/default.nix
index ea216335b7a..279c8a0fd05 100644
--- a/pkgs/development/libraries/nlopt/default.nix
+++ b/pkgs/development/libraries/nlopt/default.nix
@@ -27,6 +27,11 @@ stdenv.mkDerivation rec {
     "OCT_INSTALL_DIR=$(out)/${octave.sitePath}/oct"
   ];
 
+  postFixup = ''
+    substituteInPlace $out/lib/cmake/nlopt/NLoptLibraryDepends.cmake --replace \
+      'INTERFACE_INCLUDE_DIRECTORIES "''${_IMPORT_PREFIX}/' 'INTERFACE_INCLUDE_DIRECTORIES "'
+  '';
+
   meta = {
     homepage = "https://nlopt.readthedocs.io/en/latest/";
     description = "Free open-source library for nonlinear optimization";
diff --git a/pkgs/development/libraries/pagmo2/default.nix b/pkgs/development/libraries/pagmo2/default.nix
index abef3d6cd1d..dfb70ab624e 100644
--- a/pkgs/development/libraries/pagmo2/default.nix
+++ b/pkgs/development/libraries/pagmo2/default.nix
@@ -5,21 +5,22 @@
 , nlopt
 , ipopt
 , boost
+, tbb
 }:
 
 stdenv.mkDerivation rec {
   pname = "pagmo2";
-  version = "2.9";
+  version = "2.18.0";
 
   src = fetchFromGitHub {
      owner = "esa";
      repo = "pagmo2";
      rev = "v${version}";
-     sha256 = "0al2i59m5qr83wz5n5408zvys0b3mc40rszf0l5b9a0gp1axj400";
+     sha256 = "0rd8scs4hj6qd8ylmn5hafncml2vr4fvcgm3agz3jrvmnc7hadrj";
   };
 
   nativeBuildInputs = [ cmake ];
-  buildInputs = [ eigen nlopt ipopt boost ];
+  buildInputs = [ eigen nlopt ipopt boost tbb ];
 
   cmakeFlags = [
     "-DPAGMO_BUILD_TESTS=no"
```

**Comment by aanderse** [comment]
Thanks @markuskowa! Feel free to push directly to this `aanderse:ipopt` or I can graft these in if you don't want to push.

**Comment by markuskowa** [comment]
> Feel free to push directly to this `aanderse:ipopt` or I can graft these in if you don't want to push.

I leave it to you add the patch to the PR.

I ran nixpkgs-review earlier (without libamplsovler) and it came back okay, with the patch above.

**Review by markuskowa — APPROVED** [review]

**Comment by markuskowa** [comment]
The linux side looks okay, but darwin fails with `libampsolver`.

**Comment by aanderse** [comment]
Currently waiting on [#162146.](https://github.com/NixOS/nixpkgs/pull/162146)

**Comment by aanderse** [comment]
@GrahamcOfBorg build python39Packages.pygmo python310Packages.pygmo

**Comment by aanderse** [comment]
@GrahamcOfBorg build python39Packages.pygmo

File: pkgs/development/libraries/science/math/libamplsolver/default.nix
Comment (markuskowa) [inline pkgs/development/libraries/science/math/libamplsolver/default.nix:34]: The build seems to fail for darwin. Maybe restrict it to Linux?

Comment (aanderse) [reply pkgs/development/libraries/science/math/libamplsolver/default.nix]: Sure. It would be nice if the darwin maintainers team could look at things like these... It is an impossible battle for those of us who have no knowledge of or access to the platform.

Pushed revision. Hopefully all good now.

**Comment by aanderse** [comment]
🎉

Thanks again for all of your help!

**Comment by markuskowa** [comment]
Thanks for all the cleanup.

**Pull request merged**
